Выдачи вне нормы.

SpaceTranquility opened the pull request:
На данный момент пока не оконченный результат.

gurynych committed:
Добавлена возможность маркировки на складе

gurynych committed:
Merge branch 'feature/I-1289/T-1794' into feature/I-1289/I-1289

gurynych committed:
Merge branch 'release/2.8' into feature/I-1289/I-1289

# Conflicts:
#	Workwear/Workwear.csproj
#	Workwear/gtk-gui/gui.stetic

gurynych committed:
Добавлена печать созданных на складе штрихкодов

gurynych committed:
Добавлен журнал подменной выдачи. Не конечные изменения

gurynych committed:
Merge branch 'refs/heads/release/2.8' into feature/I-1289/T-1801

# Conflicts:
#	Workwear.Sql/Scripts/2.8.18.sql
#	Workwear.Sql/Scripts/new_empty.sql
#	Workwear/Journal/ViewModels/Stock/StockBalanceJournalViewModel.cs
#	Workwear/gtk-gui/gui.stetic

gurynych committed:
Merge remote-tracking branch 'refs/remotes/origin/release/2.8' into feature/I-1289/T-1801

gurynych committed:
Merge branch 'refs/heads/release/2.8' into feature/I-1289/T-1801

gurynych committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into feature/I-1289/T-1801

# Conflicts:
#	Workwear.Sql/Scripts/new_empty.sql

gurynych committed:
Добавлен рабочий журнал операций сверх нормы

gurynych committed:
Исправление выводимых названий

gurynych committed:
Добавлено отображение штрихкодов в таблице документа

gurynych committed:
Добавлена возможность добалять и удалять конкретные штрихкоды в документе для строк

SpaceTranquility committed:
Merge branch 'master' into feature/I-1289/T-1801

SpaceTranquility committed:
Правка интерфейса

SpaceTranquility committed:
Механиизм выдачи штрихкодов

SpaceTranquility committed:
Переименования

SpaceTranquility committed:
Сравнение параметров, правки привязок

SpaceTranquility committed:
Исправление запуска

SpaceTranquility committed:
Фильтр

SpaceTranquility committed:
Дополнение к фильтру

SpaceTranquility committed:
Скилет документа маркировки

SpaceTranquility committed:
Маркировка. Журналы, открытие, загрузка, удаление, сохранение, правки.

SpaceTranquility committed:
Фильтр и создание из журнала остатков

SpaceTranquility committed:
Печать

Gankov commented on Workwear.Sql/Scripts/2.8.18.sql:
```diff
@@ -7,9 +7,9 @@
-		`category_for_analytic_id` int unsigned null default null,
+		`category_for_analytics_id` int unsigned null default null,
 	add constraint `FK_protection_tools_category_for_analytics`
-		foreign key (`category_for_analytic_id`)
```
Почему в пулреквесте это вообще затрагиватеся? Надо убрать.

Gankov committed:
Переименовал файл верменно, чтобы не конфликтовал уже выложенным в релиз

SpaceTranquility committed:
Доделка фильтра

SpaceTranquility committed:
Merge remote-tracking branch 'origin/feature/I-1289/T-1801' into feature/I-1289/T-1801

SpaceTranquility committed:
Изменения в базе, переименования, вкладка, коллекции

SpaceTranquility committed:
Потерянная вьюшка

SpaceTranquility committed:
Фильтры в журналах, доделка разовой выдачи

SpaceTranquility committed:
Мерж из мастера

SpaceTranquility committed:
Фикс конфликтов

SpaceTranquility committed:
Исправление маркировки

Gankov commented on Workwear.Sql/Scripts/2.8.18.sql:
Почему в этом файле вообще есть исправления?

SpaceTranquility committed:
Вывод количества промаркированного

SpaceTranquility committed:
Лог и права

SpaceTranquility committed:
Исправление гостевой

SpaceTranquility committed:
Исправил гостевую

SpaceTranquility committed:
Улучшил фильтрацию

SpaceTranquility committed:
Исправил счёт промаркированного

SpaceTranquility committed:
Merge remote-tracking branch 'origin/master' into feature/I-1289/T-1801

SpaceTranquility committed:
Правка графики

Gankov commented on Workwear.Desktop/Domain/Stock.Documents/OverNorm.cs:
```diff
@@ -0,0 +1,81 @@
+		#endregion
+
+		#region Not Mapped Propertis
+		public virtual string Title => $"Выдача вне нормы ({Type.GetAttribute<DisplayAttribute>().Name}) №{(string.IsNullOrEmpty(DocNumber) ? Id.ToString() : DocNumber)} ({Type.GetAttribute<DisplayAttribute>().Name}) от {Date:d}";
```
У нас же для этого есть более простая функция GetEnumTitle

Gankov commented on Workwear.Desktop/Domain/Stock.Documents/OverNorm.cs:
```diff
@@ -0,0 +1,81 @@
+		#region Propertires
+		private IObservableList<OverNormItem> items = new ObservableList<OverNormItem>();
+		[Display (Name = "Строки документа")]
+		public virtual IObservableList<OverNormItem>Items {
```
пробел

Gankov commented on Workwear.Desktop/Domain/Stock.Documents/OverNorm.cs:
```diff
@@ -0,0 +1,81 @@
+				throw new ArgumentNullException(nameof(operation));
+			Items.Add(new OverNormItem(this, operation) { Param = param });
+		}
+		#endregion
```
мне кажется закрытие региона должно быть в другом месте.

Gankov commented on Workwear.Desktop/Domain/Stock.Documents/OverNorm.cs:
```diff
@@ -0,0 +1,81 @@
+				yield return new ValidationResult("Документ должен содержать хотя бы одну строку",
+					new[] { nameof(Items) });
+		
+			if (Items.Any(x => x.Param == null)) 
```
Может быть все нормально, просто хочу понять, это точно пользовательская проблема? Как он ее может создать.

Gankov commented on Workwear.Desktop/Domain/Stock.Documents/OverNormItem.cs:
```diff
@@ -0,0 +1,65 @@
+		}
+		
+		private OverNormOperation overNormOperation;
+		[Display(Name = "Оперерация")]
```
опечатка

Gankov commented on Workwear.Desktop/Domain/Stock/Barcode.cs:
```diff
@@ -77,6 +87,12 @@ public virtual IList<BarcodeOperation> BarcodeOperations {
 		}
+		
+////1289 проверить коментарий
+		//Предворительно нужно загрузиоть все BarcodeOperation и связанные с ними операции иначе будет много запросов в базу
```
одни опечатки

Gankov commented on Workwear.Desktop/HibernateMapping/Stock/BarcodeMap.cs:
```diff
@@ -18,14 +18,15 @@ public BarcodeMap()
 			
 			HasMany<BarcodeOperation>(x => x.BarcodeOperations)
-				.KeyColumn("barcode_id").Inverse().LazyLoad();
+				.KeyColumn("barcode_id").Inverse();
```
Мы точно в этом месте хотим убрать лейзи? Ну типа получится что мы в любом случае для штрихкода будем получать все его операции, Зачем?

SpaceTranquility replied:
Это используется для вывода дат и номеров комплектов. Скорее всего он будет почти всегда загружаться сразу. Я подумал,   что подтормаживание при открытии диалога или журнала менее бесяце чем подгрузка в процессе. Но в какой момент и по какому критерию конкретно я убрал, уже не помню, если честно

Gankov commented on Workwear.Desktop/Models/Operations/OperationToDocumentReference.cs:
```diff
@@ -61,18 +64,23 @@ public StockDocumentType? DocumentType {
 					return StockDocumentType.Completion;
 				if(InspectionId.HasValue)
 					return StockDocumentType.InspectionDoc;
+				if(BarcodingId.HasValue)
```
Тут точно ничего не перепутано? Мне очень странно почему здесь внесе документ BarcodingId но не внесен документ сверх нормы. когда логиченее наоборот. я не понимаю где может понадобится ссылка на документ баркодинга, так как он не создает складских операций.... Зачем это здесь, при это выдача сверх нормы такие операции создает.

Gankov commented on Workwear.Desktop/Tools/Barcodes/BarcodeService.cs:
```diff
@@ -78,8 +91,42 @@ public static bool CheckBarcode(string barcode, BarcodeTypes type) {
 		}
 
+		public int CountAllBarcodes(IUnitOfWork unitOfWork, StockPosition stockPosition)
+//public int CountAllBarcodes(IUnitOfWork unitOfWork, Nomenclature nomenclature, Size size = null, Size height = null) 
```
Мусор?

Gankov commented on Workwear.Desktop/Tools/OverNorms/Models/GuestModel.cs:
```diff
@@ -0,0 +1,169 @@
+				Nomenclature = param.Nomenclature,
+				WearSize = param.Size,
+				Height = param.Height
+				//TODO: owner
```
С собственником обязательно решить вопрос, это же честь стокпозиции.

Gankov commented on Workwear.Desktop/Tools/OverNorms/Models/SimpleModel.cs:
```diff
@@ -0,0 +1,134 @@
+			this.uow = uow ?? throw new ArgumentNullException(nameof(uow));
+		}
+
+//public override bool Editable { get; }
```
Может удалить везде, что оно вообще значало?

Gankov commented on Workwear.Desktop/Tools/OverNorms/Models/SubstituteFundModel.cs:
```diff
@@ -0,0 +1,182 @@
+				Amount = operation.WarehouseOperation.Amount,
+				Nomenclature = operation.WarehouseOperation.Nomenclature,
+				WearSize = operation.WarehouseOperation.WearSize,
+				Height = operation.WarehouseOperation.Height
```
Тоже нету собсвенника, а это часть сток позиции.

Gankov replied:
Вообще если честно, надо в таких ситуациях сто позицию предавать.
Нельзя в так их места отдельять их по отдельносты... Можно только в документе прихода, во всех расходных по идеи везде списываютя именно конкретные стоковые позиции. Сто позишен как объект вообзе создавался для того, чтобы можно было в сток позишен добавить еще один параметр, и не надо было хотдить по всему коду и искать где добавить еще один параметр, потому что он везде добавится автоматически.

Gankov commented on Workwear.Desktop/ConfigureDeletion.cs:
```diff
@@ -171,8 +174,13 @@ public static void ConfigureDeletion ()
-				.AddDeleteDependence<BarcodeOperation>(x => x.WarehouseOperation);
+				.AddDeleteDependence<BarcodeOperation>(x => x.WarehouseOperation)
+				.AddDeleteDependence<OverNormOperation>(x => x.WarehouseOperation);
+				//.AddDeleteDependence<OverNormItem>(x => x.OverNormOperation.WarehouseOperation);
```
Мусор?

Gankov commented on Workwear.Desktop/ConfigureDeletion.cs:
```diff
@@ -146,7 +146,10 @@ public static void ConfigureDeletion ()
-				;
+				.AddDeleteDependence<BarcodeOperation>(x => x.EmployeeIssueOperation)
+				.AddClearDependence<IssuanceSheetItem>(x => x.IssueOperation)
+				.AddDeleteDependence<InspectionItem>(x => x.OperationIssue)
```
Зачем это добавлено если оно все есть выше?

Gankov commented on Workwear.Desktop/ConfigureDeletion.cs:
```diff
@@ -306,21 +314,15 @@ public static void ConfigureDeletion ()
 				.AddClearDependence<ItemsType>(x => x.Units);
 
 			DeleteConfig.AddHibernateDeleteInfo<Nomenclature> ()
-				.AddDeleteDependence<ProtectionToolsNomenclature>(x => x.Nomenclature)
```
Почему это удаляется из номеклатуры? Думаю это какая то ошибка.

Gankov commented on Workwear.Sql/Scripts/new_empty.sql:
```diff
@@ -2116,10 +2119,160 @@ CREATE TABLE IF NOT EXISTS `operation_barcodes` (
+-- -----------------------------------------------------
+-- Table `operation_over_norm`
+-- -----------------------------------------------------
+create table if not exists `operation_over_norm`
```
Мне кажется тут if not exists нельзя допускать. Пому что если вдруг окажется что табличка есть, то не перезапишет ее.

Gankov commented on Workwear/gtk-gui/Workwear.ReportParameters.Views.StockAllWearView.cs:
```diff
@@ -60,8 +60,8 @@ protected virtual void Build()
 			this.ydateReport.Name = "ydateReport";
 			this.ydateReport.WithTime = false;
 			this.ydateReport.Date = new global::System.DateTime(0);
-			this.ydateReport.IsEditable = true;
```
Здесь бы лучше назад. Как было. В обоих параметрах, в первом просто нельзя дату редактирвать. второе не удобно.

Gankov commented on Workwear/gtk-gui/Workwear.Views.Statements.IssuanceSheetView.cs:
```diff
@@ -291,7 +291,7 @@ protected virtual void Build()
 			this.label7.Name = "label7";
 			this.label7.Xalign = 1F;
-			this.label7.LabelProp = global::Mono.Unix.Catalog.GetString("Ответствеый за передачу СИЗ:");
+			this.label7.LabelProp = global::Mono.Unix.Catalog.GetString("Ответственный за передачу СИЗ:");
```
Может лучше это исправить в исходной ветке? даже в 2.8 если там есть такая же ошибка.

Gankov commented on Workwear/gtk-gui/Workwear.Views.Stock.CollectiveExpenseView.cs:
```diff
@@ -269,7 +269,7 @@ protected virtual void Build()
 			this.label7.Name = "label7";
 			this.label7.Xalign = 1F;
-			this.label7.LabelProp = global::Mono.Unix.Catalog.GetString("Ответствеый за передачу СИЗ:");
+			this.label7.LabelProp = global::Mono.Unix.Catalog.GetString("Ответственный за передачу СИЗ:");
```
Аналогично.

Gankov commented on Workwear/Journal/Filter.ViewModels/Stock/StockBalanceFilterViewModel.cs:
```diff
@@ -76,16 +82,36 @@ public virtual bool SensitiveDate {
+		public virtual bool VisibleBarcodes => FeaturesService.Available(WorkwearFeature.Barcodes) && NomenclaturesWithBarcodes.Any();
 		private bool canChooseAmount = false;
-		public bool CanChooseAmount {
+		public virtual bool CanChooseAmount {
```
Здесь виртуал для тестов добавляется? Не совсем понял.

Gankov commented on Workwear/Journal/ViewModels/Stock/BarcodeJournalViewModel.cs:
```diff
@@ -87,7 +139,57 @@ protected override IQueryOver<Barcode> ItemsQuery(IUnitOfWork uow)
+			return query;
+		}
+		
+		#region Constraint
```
Не совсем понял как используются эти свойства? в запросе исполняются те которые в фильтре.

Gankov commented on Workwear/Journal/ViewModels/Stock/StockMovementsJournalViewModel.cs:
```diff
@@ -91,6 +91,8 @@ protected IQueryOver<WarehouseOperation> ItemsQuery(IUnitOfWork uow)
 			IssuanceSheet issuanceSheetAlias = null;
 			IssuanceSheetItem issuanceSheetItem = null;
+///1289
+//TODO  Нужно ли отображать маркировку в движениях?
```
Нет. Потому что это не движение

Gankov commented on Workwear/ViewModels/Company/EmployeeChildren/EmployeeOverNormViewModel.cs:
```diff
@@ -0,0 +1,129 @@
+			
+			var query = UoW.Session.QueryOver(() => overNormOperationAlias)
+				.Where(e => e.Employee.Id == Entity.Id);
+/*
```
Мусор?

Gankov commented on Workwear/ViewModels/Company/EmployeeChildren/EmployeeOverNormViewModel.cs:
```diff
@@ -0,0 +1,129 @@
+				.JoinAlias(() => warehouseOperationAlias.Height, () => heightAlias, JoinType.LeftOuterJoin)
+				.JoinAlias(() => nomenclatureAlias.Type, () => nomenclatureItemTypesAlias, JoinType.LeftOuterJoin)
+				.JoinAlias(() => nomenclatureItemTypesAlias.Units, () => nomenclatureUnitsAlias, JoinType.LeftOuterJoin)
+//.Where (e => e.AutoWriteoffDate == null || e.AutoWriteoffDate > DateTime.Today)
```
Мусор?

Gankov commented on Workwear/ViewModels/Stock/BarcodingViewModel.cs:
```diff
@@ -0,0 +1,301 @@
+			
+			if(Entity.Id == 0) {
+				Entity.CreatedbyUser = userService.GetCurrentUser();
+				logger.Info($"Создание Нового документа маркировки");
```
нового с маленькой буквы

Gankov commented on Workwear/CreateProjectParam.cs:
```diff
@@ -92,6 +92,8 @@
 using Workwear.Tools.Permissions;
 using Workwear.Tools.Sizes;
+using Workwear.Tools.OverNorms;
+using Workwear.Tools.OverNorms.Impl;
```
Мне не нравится Impl. давай уберем

Gankov commented on Workwear/mono_crash.17dd496f8d.0.json:
Этот файл точно лишний.

SpaceTranquility committed:
Убрал лишнее пространство и мусор

SpaceTranquility committed:
Merge branch 'master' into feature/I-1289/T-1801

SpaceTranquility committed:
Починка стетика и исправление сщетчиков баркодов

SpaceTranquility committed:
Недокомиченное

SpaceTranquility committed:
Немного подправил

SpaceTranquility committed:
Исправление склада, удаления, менюшка на кнопке

Gankov commented on Workwear/ViewModels/Stock/BarcodingViewModel.cs:
```diff
@@ -108,21 +107,37 @@ public string Total {
 			CalculateTotal();
 		}
 		
+	/*	public void DeleteFromItem(BarcodingItem item, Barcode barcode) {
```
может не надо оставлять закомменитрованный код.

SpaceTranquility committed:
Своя печатка

SpaceTranquility committed:
Добавил учёт износа и собственника в счёт промаркированного

SpaceTranquility committed:
Потенциально от дублирования

SpaceTranquility committed:
Добавил нумерации в маркировке

SpaceTranquility committed:
Ноомера в диалогах

SpaceTranquility committed:
Сдеал общий макет штрихкода

SpaceTranquility committed:
Merge remote-tracking branch 'origin/master' into feature/I-1289/T-1801

SpaceTranquility committed:
Правки от мержа и часть исправлений

SpaceTranquility committed:
Дубли в stetic

SpaceTranquility committed:
Переделал ссылки и складские движения

SpaceTranquility committed:
Лишнее

SpaceTranquility committed:
Правки выдач вне нормы

Gankov committed:
Исправление орфографических ошибок

Gankov committed:
Удалил не нужные методы

Gankov committed:
Merge branch 'master' into feature/I-1289/T-1801

Gankov committed:
Перенес изменения для выдачи вне нормы в 2.10

Gankov committed:
Исправления настроек удаления по тестам.

Gankov committed:
Исправления на основании тестов

Ваня!! Почему тесты не смотришь вообще когда готовишь изменения, ну блин, зачем мы их пишем :(

Gankov committed:
Удалил не нужный файл

Gankov committed:
Исправил ошибку во внешнем ключе, ссылался не на то поле

Как мы вообще тестировали код, не понятно, походу в базе коса этого ключа нет. Пиздос сейчас бы в релиз выкатили.

Gankov committed:
Исправлены 2 проблемы в коде по ревью ИИ

  - High: в Workwear.Desktop/Tools/OverNorms/Models/SimpleModel.cs:100 создаётся OverNormOperation, но не заполняются обязательные поля
    Nomenclature, WearSize, Height, WearPercent, Barcodes/параметры. При сохранении это конфликтует с .Not.Nullable() для nomenclature_id. Если
    этот путь реально используется, документ не сохранится корректно. Если путь мёртвый, лучше убрать/сузить API, чтобы он не выглядел рабочим.

  - High: удаление штрихкода из документа маркировки разбалансирует складские операции. В Workwear/ViewModels/Stock/BarcodingViewModel.cs:110
    уменьшается только OperationExpense.Amount, но соответствующий OperationReceipt.Amount не меняется. При частичном удалении расход и приход в
    одном документе начинают расходиться.

Gankov committed:
Убрал не нужные virtual не понимаю на кой черт они были добавлены.

Gankov committed:
Убрал if not exists  так как считаю что для миграции это вредно можно случайно пропустить изменения если таблица сущенствует программ не упадет. И изменения не применятся а мы не узнаем. это может приводить к неконсистентности базы. Например как уже произошлов в косе, нет одного улюча, а в скриптах есть.

Gankov committed:
Вернул назад лейзи загрузку, хотя судя по всему это все равно бы не помогло потому что по умолчанию колекции лейзи

Но сделал явно. на вский случай, и для ускорения во всех местах загружаю через фитч нужные данные.

Gankov committed:
Включил обновления на 2.10.13

Gankov committed:
Исправил ошибки в скриптах создания базы.

Gankov committed:
В возможности добавлена редакция Корпорация

Gankov committed:
Исправление опечаток + убрал лишнее.

Gankov committed:
Добавил разные доступы к разным видам выдачи вне нормы.

Gankov committed:
Исправление скриптов базы данных

Gankov committed:
Исправил ссылки на документацию из новых документов.

Gankov committed:
Удалил из скриптов старое забытое поле, которое осталось от старой реализации.

Gankov committed:
Очередные исправления опечаток и каких то странных не нужных кусков кода.

Gankov committed:
Удалил какую то не нужную не понятную коллекцию, которая дублирует баркод операции

И потециально усложняет код, если ее попытаться использовать. Так как происходит какое то дублирование и не понятно где правда в операциях или в списке баркодов если поменять что-то одно, или придется всегда синхронризировать их, тогда не ясно нахрена дублирование.
В обще кажется вредительская идея, потенциально приводящая к багам, более того в простой модели она уже похоже привела к багам, там из за передачи списка кодов какая то вакханалися сносящая id при обновленияих.

Gankov committed:
Исправил сборку.

Gankov committed:
Попытался сильно упростить код

Во первых убрать дублирования кода между моделями, во вторых зачем то использовалась промежуточная прослойка в хранении данных с параметрами, что приводила к необходимости лишней синхронизации между объектами.

Gankov committed:
Добавил возможность выдавать вне нормы не только маркерованную продукцию.

Gankov committed:
Исправил отрисовку нескольких шрих кодов, при добавлении

Ранее не работало из за того что события перерисовки не доходили до таблицы.

Gankov committed:
Исправление теста.

Gankov committed:
Починил кнопку удалить штрих код.

Она не работала корректно, всегда был серая даже если строка со шрих кодами выбрана.

Gankov committed:
Добавлена валидация на то что в документ не добавляется два одинаковыш штрих кода, плюс с сам объект валидацию для проверки что на складе есть нужное количество.

Gankov committed:
Переделал код чтобы сторк репозиторий мог получить юнит оф ворт из конструктора, шаг к будущим изменениями в ветке некст, и плюс это позволяет сдеалть авлидацию и тесты проще.

Gankov committed:
Добавил тест на проверку остатков на складе документом сверхнормы.

Gankov committed:
Добавил очистку документа при смене склада.

Gankov committed:
Исправлена обработка даты документа, чтобы она применяла права на изменение даты. как в других документах.

Gankov committed:
Очередное исправление мелких опечаток и удаление мусора.

Gankov committed:
Исправил разночтения по длинне лейбла в баге оно максимум 50, а вот метод почему то позволял сделать 100

Gankov committed:
Очередное исправление орфографических ошибок

Gankov committed:
Удалил не нужные не используемые методы

При чем проверил, запрос содержит не корректруню логику, через журнал выбираем свободные шрихкоды совершенно иначе. Эти методы все равно нельзя в будущем испольвзоать.

Gankov committed:
Удалил не используемый интерфейс, так как весь код все равно реализован через базовый класс интерфейс просто увеличивал кодовую базу.

Gankov committed:
Наше еще один не нужный файл.

Gankov committed:
ченджес

Gankov committed:
Исправил поломку в запросе

SpaceTranquility committed:
Включил вкладку, убрал лишние комменты

Gankov committed:
Добавил одну картинку.

SpaceTranquility committed:
Вывод меток во вкладке

Gankov committed:
Удалил информацию о редакции корпорация.

Gankov committed:
Исправил неверное использование функций программы.

Gankov committed:
Merge remote-tracking branch 'origin/feature/I-1289/T-1801' into feature/I-1289/T-1801

Gankov committed:
Убрал какой то старыный алиас.

Gankov committed:
Видимо какой то глюк мержей, строчки продублированы.

Gankov committed:
Исправлен поиск в журнале

Не соответсвие подсветки и старанные решение фильтрах, в некоторых кейсах дублирование.

SpaceTranquility committed:
Учёт новых операций при определении сотрудника носящего штрихкод

Gankov committed:
Revert "Видимо какой то глюк мержей, строчки продублированы."

This reverts commit 8af13ff66cca36b84184f0388a5d5beba69c686b.
